string input
-------------------
float32 output
